parser.c

```diff
@@ -633,20 +524,49 @@ void process_pihole_log(int file)
 				continue;
 
 			// Get dnsmasq's ID for this transaction
-			int dnsmasqID = getID(readbuffer);
+			int dnsmasqID = getdnsmasqID(readbuffer);
 			// Skip invalid lines
 			if(dnsmasqID < 0)
 				continue;
 
-			// Get ID of forward destination, create new forward destination record
-			// if not found in current data structure
-			int forwardID = getforwardID(readbuffer, true);
-			if(forwardID == -2)
+			// Get forward destination
+			// forwardstart = pointer to | in "forwarded domain.name| to www.xxx.yyy.zzz\n"
+			const char *forwardstart = strstr(readbuffer, " to ");
```

Comment:
Why was this logic broken out of `findForwardID`?

Reply:
Because we have the other similar subroutines (`findClientID()`, `findDomainID()`, `findOverTimeID()`) work on already extracted strings of log file lines. In contrast, `findForwardID()` had to be provided with a non-digested log line which was inconsistent.